ConcurrentModificationException when working with multiple sub-projects

We can see intermittent build failures on projects with multiple (4-5) sub-projects that involve CycloneDx. The stack trace might indicate a Gradle issue but it's in fact very similar to [1], so this is likely with the CycloneDX plugin.

This can be seen in Gradle 6.7. Unfortunately, I can't provide a small test project to reproduce as it only seems to happen on larger projects.

```
Caused by: java.util.ConcurrentModificationException
	at org.gradle.api.internal.DefaultDomainObjectCollection$IteratorImpl.next(DefaultDomainObjectCollection.java:479)
	at org.cyclonedx.gradle.CycloneDxTask.createBom(CycloneDxTask.java:164)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:104)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:58)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:51)
	at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:29)
```

[1] https://github.com/gradle/gradle/issues/10042

--- a/src/main/java/org/cyclonedx/gradle/CycloneDxTask.java
+++ b/src/main/java/org/cyclonedx/gradle/CycloneDxTask.java
@@ -147,10 +147,12 @@ public class CycloneDxTask extends DefaultTask {
                 .collect(Collectors.toSet());
 
         final Metadata metadata = createMetadata();
-        final Set<Component> components = getProject().getAllprojects().stream()
-            .flatMap(p -> p.getConfigurations().stream())
-            .filter(configuration -> shouldIncludeConfiguration(configuration) && !shouldSkipConfiguration(configuration) && canBeResolved(configuration))
-            .flatMap(configuration -> {
+        final Set<Configuration> configurations = getProject().getAllprojects().stream()
+                .flatMap(p -> p.getConfigurations().stream())
+                .filter(configuration -> shouldIncludeConfiguration(configuration) && !shouldSkipConfiguration(configuration) && canBeResolved(configuration))
+                .collect(Collectors.toSet());
+
+        final Set<Component> components = configurations.stream().flatMap(configuration -> {
                 final Set<Component> componentsFromConfig = Collections.synchronizedSet(new LinkedHashSet<>());
                 final ResolvedConfiguration resolvedConfiguration = configuration.getResolvedConfiguration();
                 final List<String> depsFromConfig = Collections.synchronizedList(new ArrayList<>());

